lib/client.js

```diff
@@ -115,6 +117,10 @@ class Client extends DispatcherBase {
       throw new InvalidArgumentError('unsupported keepAlive, use pipelining=0 instead')
     }
 
+    if (reset != null && reset.constructor !== Boolean) {
```

Comment:
typeof reset === 'boolean'?

Reply:
Done in [bf16edd](https://github.com/nodejs/undici/pull/1829/commits/bf16edd8db8f377257c806d88a4bfc2ff89dfdb7)